src/main/java/com/google/devtools/build/lib/rules/cpp/ShowIncludesFilter.java

```diff
@@ -54,10 +56,35 @@ public ShowIncludesFilter(String sourceFileName, String workspaceName) {
     private final ByteArrayOutputStream buffer = new ByteArrayOutputStream(4096);
     private final Collection<String> dependencies = new ArrayList<>();
     private static final int NEWLINE = '\n';
-    private static final String SHOW_INCLUDES_PREFIX = "Note: including file:";
+    // "Note: including file:" in 14 languages,
+    // cl.exe will print different prefix according to the locale configured for MSVC.
+    private static final List<String> SHOW_INCLUDES_PREFIXES = ImmutableList.of(
+        HexToUTF8String(new int[] {0x4e, 0x6f, 0x74, 0x65, 0x3a, 0x20, 0x69, 0x6e, 0x63, 0x6c, 0x75, 0x64, 0x69, 0x6e, 0x67, 0x20, 0x66, 0x69, 0x6c, 0x65, 0x3a}), // English
```

Comment:
Since we store raw data, we might as well store the signed byte array:

```java
new String(new byte[] { -26, -77, -88, -26, -124, -113, 58, 32, -27, -116, -123, -27, -112, -85, -26, -86, -108, -26, -95, -120, 58}, StandardCharsets.UTF_8),
new String(new byte[] { 80, 111, 122, 110, -61, -95, 109, 107, 97, 58, 32, 86, -60, -115, 101, 116, 110, -60, -101, 32, 115, 111, 117, 98, 111, 114, 117, 58}, StandardCharsets.UTF_8),
new String(new byte[] { 72, 105, 110, 119, 101, 105, 115, 58, 32, 69, 105, 110, 108, 101, 115, 101, 110, 32, 100, 101, 114, 32, 68, 97, 116, 101, 105, 58}, StandardCharsets.UTF_8),
new String(new byte[] { 78, 111, 116, 101, 58, 32, 105, 110, 99, 108, 117, 100, 105, 110, 103, 32, 102, 105, 108, 101, 58}, StandardCharsets.UTF_8),
new String(new byte[] { 82, 101, 109, 97, 114, 113, 117, 101, -62, -96, 58, 32, 105, 110, 99, 108, 117, 115, 105, 111, 110, 32, 100, 117, 32, 102, 105, 99, 104, 105, 101, 114, -62, -96, 58}, StandardCharsets.UTF_8),
new String(new byte[] { 78, 111, 116, 97, 58, 32, 102, 105, 108, 101, 32, 105, 110, 99, 108, 117, 115, 111}, StandardCharsets.UTF_8),
new String(new byte[] { -29, -125, -95, -29, -125, -94, 58, 32, -29, -126, -92, -29, -125, -77, -29, -126, -81, -29, -125, -85, -29, -125, -68, -29, -125, -119, 32, -29, -125, -107, -29, -126, -95, -29, -126, -92, -29, -125, -85, 58}, StandardCharsets.UTF_8),
new String(new byte[] { -20, -80, -72, -22, -77, -96, 58, 32, -19, -113, -84, -19, -107, -88, 32, -19, -116, -116, -20, -99, -68, 58}, StandardCharsets.UTF_8),
new String(new byte[] { 85, 119, 97, 103, 97, 58, 32, 119, 32, 116, 121, 109, 32, 112, 108, 105, 107, 117, 58}, StandardCharsets.UTF_8),
new String(new byte[] { 79, 98, 115, 101, 114, 118, 97, -61, -89, -61, -93, 111, 58, 32, 105, 110, 99, 108, 117, 105, 110, 100, 111, 32, 97, 114, 113, 117, 105, 118, 111, 58}, StandardCharsets.UTF_8),
new String(new byte[] { -48, -97, -47, -128, -48, -72, -48, -68, -48, -75, -47, -121, -48, -80, -48, -67, -48, -72, -48, -75, 58, 32, -48, -78, -48, -70, -48, -69, -47, -114, -47, -121, -48, -75, -48, -67, -48, -72, -48, -75, 32, -47, -124, -48, -80, -48, -71, -48, -69, -48, -80, 58}, StandardCharsets.UTF_8),
new String(new byte[] { 78, 111, 116, 58, 32, 101, 107, 108, 101, 110, 101, 110, 32, 100, 111, 115, 121, 97, 58}, StandardCharsets.UTF_8),
new String(new byte[] { -26, -77, -88, -26, -124, -113, 58, 32, -27, -116, -123, -27, -112, -85, -26, -106, -121, -28, -69, -74, 58}, StandardCharsets.UTF_8),
new String(new byte[] { 78, 111, 116, 97, 58, 32, 105, 110, 99, 108, 117, 115, 105, -61, -77, 110, 32, 100, 101, 108, 32, 97, 114, 99, 104, 105, 118, 111, 58}, StandardCharsets.UTF_8),
```

They are ascending by language ID, so English is the 4th from the top.

Reply:
Awesome! I moved English to the first one since it's the most used one.